packages/solidjs/src/errorboundary.ts

```diff
@@ -0,0 +1,31 @@
+import { captureException } from '@sentry/browser';
+import type { Component, JSX } from 'solid-js';
+import { mergeProps, splitProps, untrack } from 'solid-js';
+import { createComponent } from 'solid-js/web';
+
+type ErrorBoundaryProps = {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  fallback: JSX.Element | ((err: any, reset: () => void) => JSX.Element);
+  children: JSX.Element;
+};
+
+/**
+ * A higher-order component to wrap Solid's ErrorBoundary to capture exceptions.
+ */
+export function withSentryErrorBoundary(ErrorBoundary: Component<ErrorBoundaryProps>): Component<ErrorBoundaryProps> {
```

Comment:
I have a naive question (sorry if I missed convos around this): Why do users need to pass in the `ErrorBoundary` component from solid here? Can we not import it like the other solid imports?

Reply:
Not naive at all. That's the approach I initially took, but it doesn't quite do the job correctly. 

From local testing and similar experience with the router instrumentation, certain functions and functionality are app context specific. For the router, this meant that we couldn't use `use` hooks because they didn't have the user's router context. For the `ErrorBoundary` the `reset` function is contextual, so exporting a generic error boundary from this package would mean the `reset` function does nothing.

Hope this helps to understand it a bit better.